Tutorial1-1Basics/src/main/java/com/smarttoolfactory/tutorial1_1basics/chapter2_material_widgets/Tutorial2_9_1SideNavigationScaffold.kt

```diff
@@ -42,33 +45,51 @@ import androidx.compose.ui.res.painterResource
 import androidx.compose.ui.text.font.FontWeight
 import androidx.compose.ui.tooling.preview.Devices
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
 import androidx.compose.ui.unit.dp
 import androidx.compose.ui.unit.sp
 import androidx.navigation.compose.NavHost
 import androidx.navigation.compose.composable
 import androidx.navigation.compose.rememberNavController
 import com.smarttoolfactory.tutorial1_1basics.R
+import com.smarttoolfactory.tutorial1_1basics.isInPreview
 import com.smarttoolfactory.tutorial1_1basics.ui.ComposeTutorialsTheme
 import com.smarttoolfactory.tutorial1_1basics.ui.components.DrawerButton
 import kotlinx.coroutines.launch
 
 @Preview
 @Composable
-fun Tutorial2_9Screen1() {
-    TutorialContent()
+fun Tutorial2_9Screen1(
+    @PreviewParameter(DrawerStateProvider::class)
+    drawerValue: DrawerValue
+) {
+    TutorialContent(
+        scaffoldState = rememberScaffoldState(
+            drawerState = rememberDrawerState(initialValue = drawerValue)
+        )
+    )
+}
+
+private class DrawerStateProvider: PreviewParameterProvider<DrawerValue> {
+    override val values: Sequence<DrawerValue>
+        get() = sequenceOf(
+            DrawerValue.Closed,
+            DrawerValue.Open
+        )
 }
 
 @Composable
-private fun TutorialContent() {
+private fun TutorialContent(scaffoldState: ScaffoldState = rememberScaffoldState()) {
```

Comment:
Hoisted `scaffoldState` to call-site where we can provide custom initial value for `DrawerState`.